components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/domain.jsp

```diff
@@ -15,23 +15,21 @@
   ~ specific language governing permissions and limitations
   ~ under the License.
   --%>
-
-<%@ page import="org.wso2.carbon.identity.application.authentication.endpoint.util.CharacterEncoder"%>
+<%@ page import="org.owasp.encoder.Encode" %>
 <%
     String errorMessage = "Authentication Failed";
     boolean loginFailed = false;
-    
-    if (CharacterEncoder.getSafeText(request.getParameter("authFailure")) != null && "true"
-    .equals(CharacterEncoder.getSafeText(request.getParameter("authFailure")))) {
-	    loginFailed = true;
-	    
-	    if(CharacterEncoder.getSafeText(request.getParameter("authFailureMsg")) != null){
-	        errorMessage = CharacterEncoder.getSafeText(request.getParameter("authFailureMsg"));
-	        
-	        if (errorMessage.equalsIgnoreCase("domain.unknown")) {
-	        	errorMessage = "Domain cannot be identified! Please retry.";
-	        }
-	    }
+
+    if (Boolean.parseBoolean(request.getParameter("authFailure"))) {
+        loginFailed = true;
+
+        if (request.getParameter("authFailureMsg") != null) {
+            errorMessage = request.getParameter("authFailureMsg");
+
+            if (errorMessage.equalsIgnoreCase("domain.unknown")) {
```

Comment:
although "errorMessage" is checked for null above, better to use "if(constant.equalsIgnoreCase(variable)" pattern as a best practice.